### ISSUE TITLE
the training loss in nan

### ISSUE THREAD

**Issue by DanMerry** [sangaram/wav2vec2#1]
![image](https://github.com/user-attachments/assets/44fd88a1-cb65-4840-96ed-ea7d20770f07)
# The training loss is always nan, how to deal with it

**Comment by sangaram** [comment]
Thank you for raising the issue @DanMerry , I will find out what's wrong and fix it. But in order to work with Wav2Vec 2.0, I highly recommend you to use the implementation available on Hugging Face:
- https://huggingface.co/facebook/wav2vec2-base-960h
- https://huggingface.co/docs/transformers/tasks/asr

**Comment by DanMerry** [comment]
Thank you for your response. The issue seems to be the presence of NaNs in the input data. After removing the NaNs, a new problem has arisen: the loss does not seem to be decreasing. How should I address this issue?
![image](https://github.com/user-attachments/assets/78b2b6b1-4a65-45a7-8cff-29b92eee6752)
